client/pages/trip-todo.jsx

```diff
@@ -113,15 +127,33 @@ function ToDoList(props) {
 }
 
 function ToDoItem(props) {
+  const [completeStatus, setCompleted] = useState(props.todo.completed);
+
+  useEffect(() => {
+    const newStatus = { completed: completeStatus };
+    const requestOptions = {
+      method: 'PATCH',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(newStatus)
+    };
+
+    fetch(`/api/triptodo/${props.todo.todoId}`, requestOptions)
+      .then(response => response.json())
+      .catch(err => console.error(err));
+  });
```

Comment:
I applaud your exploration into hooks. However, `useEffect` runs its callback on every re-render, so this component is spamming your route every time it re-renders, which is really bad for performance. Once you have moved your fetch into the list component, you can delete both of these hooks. I definitely recommend refactoring your app to hooks instead of classes once it's done so you can finish learning them.